### DISCUSSION TITLE
Consumer runs into an error: The AMQP operation was interrupted: AMQP close-reason, initiated by Peer, code=406, text='PRECONDITION_FAILED - delivery acknowledgement on channel 2 timed out. Timeout value used: 3600000 ms

### DISCUSSION THREAD

**Discussion by EmanueleBaron** [rabbitmq/rabbitmq-server#12864] [Other] [answered]
### Describe the bug

I'm receved this error "Already closed: The AMQP operation was interrupted: AMQP close-reason, initiated by Peer, code=406, text='PRECONDITION_FAILED - delivery acknowledgement on channel 2 timed out. Timeout value used: 3600000 ms"
This occurs when my queue has too many messages to process. 
I'm using Rabbitmq version 7, I've already changed the channel timeout from 1800000 ms to 3600000 ms, but it didn't help. What's wrong in my code? 

### Reproduction steps

    public static async Task ConsumeQueue(IConnection conn, string queueName, TaskCompletionSource tcs, ILogger logger)
    {
        using var channel = await conn.CreateChannelAsync();
        var consumer =  new AsyncEventingBasicConsumer(channel);

        consumer.ReceivedAsync += async (co, ea) =>
        {
            try
            {
                var body = ea.Body.ToArray();
                var message = Encoding.UTF8.GetString(body);
                string apiResponse = await SendEmail.SendAPIMensageiroAsync(message, logger);

                await channel.BasicAckAsync(ea.DeliveryTag, multiple: false);

                if (apiResponse.Contains("Enviado"))
                {                   
                    await Publish.PublishMessage(conn, "EmailEnviado", $"{queueName}Enviados", message);
                }
                else
                {
                    message = message.Replace("},", $"}}, \"StatusEnvioErro\": \"{apiResponse}\" ,");
                   await Publish.PublishMessage(conn, "EmailErro", $"{queueName}Erro", message);
                }

                await Task.Delay(100);

            }
            catch (Exception ex)
            {
                logger.LogError($"Erro: {ex.Message}");
            }
        };

       await channel.BasicConsumeAsync(queueName, false, consumer);
        await tcs.Task;
    }
}

### Expected behavior

What should I do for resolved this error?

**Comment by mkuratczyk** [comment]
It is a bug in your code - you are not acknowledging messages in some situations apparently.

GH issues for RabbitMQ is not the right place for ask for help debugging your code. If you want help from the community, create a minimal, executable, self-contained application that exhibits this behaviour, put it on github and ask on stack overflow for example.

Lastly, there is no such thing as "rabbit 7". The latest version is 4.0.

**Reply by michaelklishin** [reply]
I suspect that it means the 7.0 series of the .NET client, which has shipped recently.

**Comment by michaelklishin** [comment, marked as answer]
@EmanueleBaron you are looking at a [Delivery acknowledgement timeout](https://www.rabbitmq.com/docs/consumers#acknowledgement-timeout). The number of messages to process is unlikely to be relevant, although a high load on a consumer can be a contributing factor when the timeout is overridden to a low value (say, a minute).

The default is 30 minutes so as @mkuratczyk has already explained, your application does not [acknowledge deliveries](https://www.rabbitmq.com/docs/consumers#acknowledgement-modes) for one reason or another. This is not a bug in RabbitMQ.
